src/Xrefcheck/Scan.hs

```diff
@@ -170,88 +168,84 @@ readDirectoryWith mode config scanner root =
       RdmBothTrackedAndUtracked -> liftA2 (<>) getTrackedFiles getUntrackedFiles
 
     getTrackedFiles = readCreateProcess
-      (shell "git ls-files"){cwd = Just root} ""
+      (shell "git ls-files"){cwd = Just $ unCanonicalPath root} ""
     getUntrackedFiles = readCreateProcess
-      (shell "git ls-files --others --exclude-standard"){cwd = Just root} ""
+      (shell "git ls-files --others --exclude-standard"){cwd = Just $ unCanonicalPath root} ""
 
-    scanFile :: FilePath -> IO (FilePath, a)
-    scanFile = sequence . (normaliseWithNoTrailing &&& scanner)
+    scanFile :: CanonicalPath -> IO (CanonicalPath, a)
+    scanFile c = (c,) <$> scanner c
 
-    isIgnored :: FilePath -> Bool
+    isIgnored :: CanonicalPath -> Bool
     isIgnored = matchesGlobPatterns root $ ecIgnore config
 
-    -- Strip leading "." and trailing "/"
-    location :: FilePath
-    location =
-      if root `equalFilePath` "."
-        then ""
-        else dropTrailingPathSeparator root
-
 scanRepo
   :: MonadIO m
```

Comment:
For the moment I have kept the `FilePath` parameters of `IO` actions as they are instead of changing them to `CanonicalPath`. They can be canonicalised inside the `IO` action and the exported interface remains the same, so I am not forced to modify the tests meanwhile.